source/fe/fe_system.cc

```diff
@@ -2334,12 +2334,6 @@ compare_for_face_domination (const FiniteElement<dim,spacedim> &fe_other) const
           domination = domination & base_domination;
         }
 
-      // if we've gotten here, then we've either found a winner or either
-      // element is fine being dominated
-      Assert (domination !=
-              FiniteElementDomination::neither_element_dominates,
-              ExcInternalError());
```

Comment:
I think this change is correct in itself -- if we can't find a winner, we should simply say so and let the calling site figure out what to do with it.